File: specialist-service/src/main/java/com/fyh/specialistservice/dto/SpecialistAdminDto.java
package com.fyh.specialistservice.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Timestamp;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SpecialistAdminDto {
    private Long id;
    private Long idUtilizator;
    private String numeUtilizator;
    private String emailUtilizator;
    private String tipUtilizator;

    private Long specializareId;
    private String denumireSpecializare;

    private List<ServiciuDto> servicii;
    private String atestat;
    private String descriere;
    private String statusValidare;
    private String soldAcumulat;
    private Long idAdmin;
    private Timestamp dataValidare;
}

File: frontend/src/app/models/specialist-admin.dto.ts
import { ServiciuDto } from './serviciu.dto';

export interface SpecialistAdminDto {
  id: number;
  idUtilizator: number;
  numeUtilizator: string;
  emailUtilizator: string;
  tipUtilizator: string;

  specializareId: number;
  denumireSpecializare: string;

  servicii: ServiciuDto[];
  atestat: string;
  descriere: string;
  statusValidare: string;
  soldAcumulat: string;
  idAdmin: number | null;
  dataValidare: string | null;
}
